### DISCUSSION TITLE
Change the default issuer for dex

### DISCUSSION THREAD

**Discussion by AnkitaBarbi** [dexidp/dex#1987] [Q&A] [answered]
I have just started working on dex , and have some concerns that I am unable to figure out

By default when we run dex issuer happens to be
```
issuer: http://127.0.0.1:5556/dex
```
but lets. say if I want the issuer to be registered with
```
issuer: https://abc.example.in
```
How do I achieve it?

**Comment by nabokihms** [comment, marked as answer]
Hello! You can change it via the dex configuration file.
There is an example of a config file https://github.com/dexidp/dex/blob/master/config.yaml.dist#L4

The `issuer` setting is what you are searching for.

**Reply by AnkitaBarbi** [reply]
hey @nabokihms  I tried that and changed in config.yaml, and also change the same for example app main.go to test it. But then it throws below error
```
failed to query provider "https://abc.example.in": oidc: failed to decode provider discovery object: expected Content-Type = application/json, got "text/html; charset=utf-8": invalid character '<' looking for beginning of value
```
What am I missing?

**Reply by nabokihms** [reply]
I tested it locally using the current master branch.
1. Run commands to make binaries:
```bash
make bin/dex examples
```
2. Change issuer in examples/config-dev.yaml to `http://abc.example.in:5556`
3. Run dex:
```bash
bin/dex serve examples/config-dev.yaml
```
4. Run example app:
```bash
bin/example-app --issuer http://abc.example.in:5556
```
5. Login via the mock connector, inspect the received token:
```json
{
  "iss": "http://abc.example.in:5556",
  "sub": "Cg0wLTM4NS0yODA4OS0wEgRtb2Nr",
  "aud": "example-app",
  "exp": 1613125917,
  "iat": 1613039517,
  "at_hash": "Edjvu-8C9X8uJDGTht3LJQ",
  "c_hash": "bRKabLB8wzx-wlDvBKKcaQ",
  "email": "kilgore@kilgore.trout",
  "email_verified": true,
  "name": "Kilgore Trout"
}
```
